examples/src/test/scala/docs/guide/ShoppingCartApp.scala

```diff
@@ -0,0 +1,157 @@
+
+//#guideSourceProviderImports
+
+//#guideProjectionImports
+
+import java.time.temporal.ChronoUnit
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Success
+
+import akka.Done
+import akka.projection.ProjectionId
+import akka.projection.ProjectionBehavior
+import akka.projection.scaladsl.Handler
+import akka.projection.cassandra.scaladsl.CassandraProjection
+import akka.stream.alpakka.cassandra.scaladsl.CassandraSessionRegistry
+import akka.stream.alpakka.cassandra.scaladsl.CassandraSession
+import org.slf4j.LoggerFactory
+
+//#guideProjectionImports
+
+//#guideSetup
+object ShoppingCartEvents {
+
+  /**
+   * This interface defines all the events that the ShoppingCart supports.
+   */
+  sealed trait Event {
+    def cartId: String
+  }
+
+  final case class ItemAdded(cartId: String, itemId: String, quantity: Int) extends Event
+  final case class ItemRemoved(cartId: String, itemId: String) extends Event
+  final case class ItemQuantityAdjusted(cartId: String, itemId: String, newQuantity: Int) extends Event
+  final case class CheckedOut(cartId: String, eventTime: Instant) extends Event
+}
+
+object ShoppingCartApp extends App {
+  private val system = ActorSystem[Nothing](Behaviors.empty, "ShoppingCartApp")
+  //#guideSetup
+  //#guideSourceProviderSetup
+  val europeanShoppingCartsTag = "carts-eu"
+  val sourceProvider: SourceProvider[Offset, EventEnvelope[ShoppingCartEvents.Event]] =
+    EventSourcedProvider
+      .eventsByTag[ShoppingCartEvents.Event](
+        system,
+        readJournalPluginId = CassandraReadJournal.Identifier,
+        tag = europeanShoppingCartsTag)
+  //#guideSourceProviderSetup
+  //#guideSetup
+
+  //#guideProjectionSetup
+  // FIXME: use a custom dispatcher for repo?
+  implicit val ec = system.classicSystem.dispatcher
+  val session = CassandraSessionRegistry(system).sessionFor("akka.projection.cassandra.session-config")
+  val repo = new DailyCheckoutProjectionRepository(session)
+  val projection = CassandraProjection.atLeastOnce(
+    projectionId = ProjectionId("shopping-carts", europeanShoppingCartsTag),
+    sourceProvider,
+    handler = () => new ShoppingCartProjectionHandler(europeanShoppingCartsTag, system, repo))
+
+  Behaviors.setup[Nothing] { context =>
+    context.spawn(ProjectionBehavior(projection), "daily-count-projection")
+    Behaviors.empty
+  }
+  //#guideProjectionSetup
+}
+//#guideSetup
+
+//#guideProjectionRepo
+class DailyCheckoutProjectionRepository(session: CassandraSession)(implicit val ec: ExecutionContext) {
+  val keyspace = "projections"
+  val table = "daily_checkouts"
+
+  def save(cartId: String, date: Instant): Future[Done] = {
+    session.executeWrite(s"INSERT INTO $keyspace.$table (cart_id, date) VALUES (?, ?)", cartId, date)
+  }
+
+  def countForDate(date: Instant): Future[Option[Int]] = {
+    session.selectOne(s"SELECT count(card_id) FROM $keyspace.$table WHERE date = ?", date).map {
```

Comment:
I don't think this query can run efficiently with that table definition. In general the first column in primary key (the partitioning key) must be included in the WHERE. It must probably be defined as `PRIMARY KEY (date, cart_id)`.

typo: card_id